crates/telio-proxy/src/proxy.rs

```diff
@@ -187,10 +208,61 @@ impl StateEgress {
 
         self.wg_addr = wg_port.map(|p| (Ipv4Addr::LOCALHOST, p).into());
 
-        self.sockets = sockets;
+        let old_sockets = self.sockets.clone();
+        self.sockets = sockets
+            .clone()
+            .into_iter()
+            .map(|(pk, socket)| {
+                (
+                    pk,
+                    (
+                        socket,
+                        old_sockets
+                            .get(&pk)
+                            .map(|(_, muted)| *muted)
+                            .unwrap_or(false),
+                    ),
+                )
+            })
+            .collect();
```

Comment:
Too much cloning IMHO
```suggestion
        let old_sockets = std::mem::take(&mut self.sockets);
        self.sockets = sockets
            .into_iter()
            .map(|(pk, socket)| {
                (
                    pk,
                    (
                        socket,
                        old_sockets
                            .get(&pk)
                            .map_or(false, |(_, muted)| *muted),
                    ),
                )
            })
            .collect();
```

Reply:
look's good